pandas/core/internals/blocks.py

```diff
@@ -793,7 +804,9 @@ def replace_list(
 
         # Exclude anything that we know we won't contain
         pairs = [
-            (x, y) for x, y in zip(src_list, dest_list) if self._can_hold_element(x)
+            (x, y)
+            for x, y in zip(src_list, dest_list)
+            if (self._can_hold_element(x) or (self.dtype == "string" and is_re(x)))
```

Comment:
Not necessarily, because if you call `_can_hold_element` in other context (eg to know if you can do a setitem like operation like fillna), an `re` object is not something you can actually set in the array. So this is somewhat specific to `replace` (I was first thinking to add a `method` keyword to `can_hold_element` and then handle it there if `method="replace"`, but in the end decided to just handle it here)

Reply:
Ah OK thanks for the explanation